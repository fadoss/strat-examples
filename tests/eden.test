semantics/eden/examples

srew [1] eval(fib2(2)) using start .


