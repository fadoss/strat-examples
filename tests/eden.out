==========================================
srewrite [1] in MAIN : eval(fib2(2)) using start .

Solution 1
rewrites: 162309
result Process: 
< "main",
  s("main") |- I -> n(2) + 
  s("main.38") |- I -> \ s("x") . (let s("x.31") = (let s("x.1") = (s("x") - s(
    "x.2")) s("x.2") = n(2) in s("x.1")) s("x.32") = (let s("x.16") = (let s(
    "x.3") = (s("x") - s("x.4")) s("x.4") = n(1) in s("x.3")) s("x.17") = n(1)
    s("x.18") = (let s("x.13") = (let s("x.11") = (s("main.38") s("x.12")) s(
    "x.12") = (let s("x.9") = (s("x") - s("x.10")) s("x.10") = n(2) in s(
    "x.9")) in s("x.11")) s("x.14") = (let s("x.7") = (s("main.38") s("x.8"))
    s("x.8") = (let s("x.5") = (s("x") - s("x.6")) s("x.6") = n(1) in s("x.5"))
    in s("x.7")) s("x.15") = (s("x.14") + s("x.13")) in s("x.15")) s("x.19") =
    if0(s("x.16"), s("x.17"), s("x.18")) in s("x.19")) s("x.33") = (let s(
    "x.28") = s("x.21") s("x.20") = (let s("x.22") = (s("x") - s("x.23")) s(
    "x.23") = n(1) in s("x.22")) s("x.21") = (s("main.38") # s("x.20")) s(
    "x.29") = s("x.25") s("x.24") = (let s("x.26") = (s("x") - s("x.27")) s(
    "x.27") = n(2) in s("x.26")) s("x.25") = (s("main.38") # s("x.24")) s(
    "x.30") = (s("x.28") + s("x.29")) in s("x.30")) s("x.34") = if0(s("x.31"),
    s("x.32"), s("x.33")) in s("x.34")) + 
  s("main.39") |- I -> n(2) + 
  s("main.40") |- I -> n(2) + 
  s("main.41") |- I -> n(2) + 
  s("main.42") |- I -> n(0) + 
  s("main.43") |- I -> n(2) + 
  s("main.44") |- I -> let s("x.28") = s("x.21") s("x.20") = (let s("x.22") = (
    s("main.41") - s("x.23")) s("x.23") = n(1) in s("x.22")) s("x.21") = (s(
    "main.38") # s("x.20")) s("x.29") = s("x.25") s("x.24") = (let s("x.26") =
    (s("main.41") - s("x.27")) s("x.27") = n(2) in s("x.26")) s("x.25") = (s(
    "main.38") # s("x.24")) s("x.30") = (s("x.28") + s("x.29")) in s("x.30") + 
  s("main.45") |- I -> n(2) + 
  s("main.46") |- I -> n(0) + 
  s("main.47") |- I -> n(2) + 
  s("main.48") |- I -> n(1) + 
  s("main.49") |- I -> n(1) + 
  s("main.50") |- I -> n(2) + 
  s("main.51") |- I -> n(2) + 
  s("main.52") |- I -> n(1) + 
  s("main.53") |- I -> n(1) + 
  s("main.54") |- I -> n(1) + 
  s("main.55") |- I -> n(1) + 
  s("main.56") |- I -> n(2) + 
  s("main.57") |- I -> n(1) + 
  s("main.58") |- I -> n(0) + 
  s("main.59") |- I -> n(1) + 
  s("main.60") |- I -> n(1) + 
  s("main.61") |- I -> n(0) + 
  s("main.62") |- I -> n(1) + 
  s("main.63") |- I -> let s("x.28") = s("x.21") s("x.20") = (let s("x.22") = (
    s("main.58") - s("x.23")) s("x.23") = n(1) in s("x.22")) s("x.21") = (s(
    "main.38") # s("x.20")) s("x.29") = s("x.25") s("x.24") = (let s("x.26") =
    (s("main.58") - s("x.27")) s("x.27") = n(2) in s("x.26")) s("x.25") = (s(
    "main.38") # s("x.24")) s("x.30") = (s("x.28") + s("x.29")) in s("x.30") + 
  s("main.64") |- I -> n(1) + 
  s("main.65") |- I -> n(0) + 
  s("main.66") |- I -> n(1) + 
  s("main.67") |- I -> let s("x.28") = s("x.21") s("x.20") = (let s("x.22") = (
    s("main.60") - s("x.23")) s("x.23") = n(1) in s("x.22")) s("x.21") = (s(
    "main.38") # s("x.20")) s("x.29") = s("x.25") s("x.24") = (let s("x.26") =
    (s("main.60") - s("x.27")) s("x.27") = n(2) in s("x.26")) s("x.25") = (s(
    "main.38") # s("x.24")) s("x.30") = (s("x.28") + s("x.29")) in s("x.30") + 
  s("main.68") |- I -> n(1) + 
  s("main.69") |- I -> n(0) + 
  s("main.70") |- I -> n(2) + 
  s("main.71") |- I -> n(0) + 
  s("main.72") |- I -> n(2) + 
  s("main.73") |- I -> n(0) + 
  s("main.74") |- I -> n(2) + 
  s("main.75") |- I -> n(1) + 
  s("main.76") |- I -> n(1) + 
  s("main.77") |- I -> n(0) + 
  s("main.78") |- I -> n(1) + 
  s("main.79") |- I -> let s("x.13") = (let s("x.11") = (s("main.38") s(
    "x.12")) s("x.12") = (let s("x.9") = (s("main.58") - s("x.10")) s("x.10") =
    n(2) in s("x.9")) in s("x.11")) s("x.14") = (let s("x.7") = (s("main.38")
    s("x.8")) s("x.8") = (let s("x.5") = (s("main.58") - s("x.6")) s("x.6") =
    n(1) in s("x.5")) in s("x.7")) s("x.15") = (s("x.14") + s("x.13")) in s(
    "x.15") + 
  s("main.80") |- I -> n(1) + 
  s("main.81") |- I -> n(0) + 
  s("main.82") |- I -> n(1) + 
  s("main.83") |- I -> let s("x.13") = (let s("x.11") = (s("main.38") s(
    "x.12")) s("x.12") = (let s("x.9") = (s("main.60") - s("x.10")) s("x.10") =
    n(2) in s("x.9")) in s("x.11")) s("x.14") = (let s("x.7") = (s("main.38")
    s("x.8")) s("x.8") = (let s("x.5") = (s("main.60") - s("x.6")) s("x.6") =
    n(1) in s("x.5")) in s("x.7")) s("x.15") = (s("x.14") + s("x.13")) in s(
    "x.15") + 
  s("main.84") |- I -> n(1) + 
  s("main.85") |- I -> n(0) + 
  s("main.86") |- I -> n(1) + 
  s("main.87") |- I -> n(0) + 
  s("main.88") |- I -> n(1), 0, 88 >
Bye.
